js/parts/Series.js

```diff
@@ -3009,13 +3009,16 @@ H.Series = H.seriesType(
                 userOptions = chart.userOptions || {},
                 userPlotOptions = userOptions.plotOptions || {},
                 typeOptions = plotOptions[this.type],
+                seriesUserOptions = merge(itemOptions),
                 options,
                 zones,
                 zone,
                 styledMode = chart.styledMode;
 
+            fireEvent(this, 'setOptions', { userOptions: seriesUserOptions });
```

Comment:
What is the reason of calling setOptions?

Reply:
The `setOptions` event was added to do what is described in https://github.com/highcharts/highcharts/pull/9976#issue-247001601 at the section Series.setOptions. The event is used at [modules/treemap.src.js#L658](https://github.com/highcharts/highcharts/blob/e9e1328cadb948a7eaa0e31533de094525ebe26d/js/modules/treemap.src.js#L658)